lib/virt_autotest/utils.pm

```diff
@@ -105,23 +105,56 @@ sub is_xen_host {
     return get_var("XEN") || check_var("SYSTEM_ROLE", "xen") || check_var("HOST_HYPERVISOR", "xen") || check_var("REGRESSION", "xen-hypervisor");
 }
 
-#check host to make sure it works well
+#check kernels by grep keywords from journals
 #welcome everybody to extend this function
-sub check_host {
+sub check_failures_in_journal {
+    my $machine //= 'localhost';
+
+    my $all_failures = "";
+    my @warnings = ('Started Process Core Dump', 'Call Trace');
+    foreach my $warn (@warnings) {
+        my $cmd = "journalctl | grep '$warn'";
+        $cmd = "ssh root\@$machine " . "\"$cmd\"" if $machine ne 'localhost';
+        if (script_run($cmd) == 0) {
+            if (get_var('KNOWN_KERNEL_BUG')) {
+                my $failure = "$warn in journals on $machine: " . get_var('KNOWN_KERNEL_BUG');
+                $all_failures .= $failure;
+                record_soft_failure("$failure (Please ignore the 'fake bsc#5555' that has to be written here due to the silly PR code rule)");
```

Comment:
Sorry to give this late comment. 
But here, in the logic of if KNOWN_KERNEL_BUG, it is assumed that for each $warn that is found, it is the same as known issue KNOWN_KERNEL_BUG, which is not correct, I am afraid, IMHO. It is fine to not do further check whether it is the same with KNOWN_KERNEL_BUG, but in the record_soft_failure, we should suggest reviewers to check by themselves. 

How about below?
```
  foreach my $warn (@warnings) {
        my $cmd = "journalctl | grep '$warn'";
        $cmd = "ssh root\@$machine " . "\"$cmd\"" if $machine ne 'localhost';
        if (script_run($cmd) == 0) {
            $all_failures .= "\"$warn\" in journals on $machine \n";
        }
  }

  if ($all_failures ne '') {
    if (get_var('KNOWN_KERNEL_BUG')) {
      record_soft_failure("Found failures in journal log", "$all_failures" . "\nThere are known kernel bugs " . get_var('KNOWN_KERNEL_BUG') . ". Please analyze journal logs to double check if new bug needs to be opened or it is old issue.");
    }
    else {
      record_soft_failure("Found new failures in journal log: fake bsc#5555(by PR rule)", "$all_failures" . "\nThis is an unknown new failure which need to be investigated!");
    }
  }
  return $all_failures;
```

Reply:
it is ok except this statement:
`record_soft_failure("Found failures in journal log", "$all_failures" . "\nThere are known kernel bugs " . get_var('KNOWN_KERNEL_BUG')`

the silly 'fake bsc#555' has to be added here to pass the CI code rule.

Reply:
And I am going to rename `KNOWN_KERNEL_BUG` to `KNOWN_KERNEL_BUGS`, a 's' is added.

Reply:
> it is ok except this statement: `record_soft_failure("Found failures in journal log", "$all_failures" . "\nThere are known kernel bugs " . get_var('KNOWN_KERNEL_BUG')`
> 
the silly 'fake bsc#555' has to be added here to pass the CI code rule.

And `record_soft_failure([$reason]);` only has one arguments.

Reply:
> the silly 'fake bsc#555' has to be added here to pass the CI code rule.

KNOWN_KERNEL_BUG should have given bugs in format bsc#xxxx,bsc#xxx? Enough? I do not know detailed requirement on this rule. But please remove 'silly', which is too sensitive ;-)

Reply:
> > the silly 'fake bsc#555' has to be added here to pass the CI code rule.
> 
> KNOWN_KERNEL_BUG should have given bugs in format bsc#xxxx,bsc#xxx? Enough? I do not know detailed requirement on this rule. But please remove 'silly', which is too sensitive ;-)

Aha, I think I got the point, in static PR CI check, there is no value yet. Indeed silly, but please exclude the word 

> And record_soft_failure([$reason]); only has one argument.
Yep, corrected in the suggestion. I guess you are quoting old comment. I was using it like record_info initially, sorry!